### ISSUE TITLE
Multiple children on a node

### ISSUE THREAD

**Issue by JakeWharton** [cashapp/redwood#11]
Notes from meeting in early January:
```
Components with multiple lists of children
  TLDR change:
    Currently we emit children directly inside the emit() call,

      emit(
        ...,
        children = {
          children()
        },
      )

    For each child group, change that to emit an intermediate Children node

      emit(
        ...,
        children = {
          Children(id = 1) {
            children1()
          }
          Children(id = 2) {
            children2()
          }
        },
      )

    Erase the Children box at the protocol layer by using its ID as the child index.

     class Screen {
       headerChildren: List<Any>
       bodyChildren: List<Any>
     }

     We want to rewrite this as if it were coded as

     class Screen {
       headerChildren: HeaderChildrenBox
       bodyChildren: BodyChildrenBox
     }
     class HeaderChildrenBox {
       list: List<Any>
     }
     class BodyChildrenBox {
       list: List<Any>
     }

     (this should be at the wire-protocol level only, with some affordances for receiving diffs)
```
